MergerService/src/run.cs

```diff
@@ -0,0 +1,166 @@
+using MergerLogic.Batching;
+using MergerLogic.DataTypes;
+using MergerLogic.ImageProccessing;
+using MergerLogic.Utils;
+using MergerService.Controllers;
+using System.Diagnostics;
+
+namespace MergerService.Src
+{
+    public class Run
+    {
+        private static List<Data> BuildDataList(Source[] paths, int batchSize)
+        {
+            List<Data> sources = new List<Data>();
+
+            if (paths.Length != 0)
+            {
+                sources.Add(Data.CreateDatasource(paths[0].Type, paths[0].Path, batchSize, true));
+                foreach (Source source in paths.Skip(1))
+                {
+                    sources.Add(Data.CreateDatasource(source.Type, source.Path, batchSize, source.IsOneXOne()));
+                }
+            }
+
+            return sources;
+        }
+
+        public static void Start()
+        {
+            Stopwatch stopWatch = new Stopwatch();
+            TimeSpan ts;
+
+            while (true)
+            {
+                // Sleep for 0.5 sec so we won't send too many requests if there are no tasks available
+                Thread.Sleep(500);
+
+                MergeTask? task = MergeTask.GetTask();
+
+                // Guard clause in case there are no batches or sources
+                if (task == null || task.Batches == null || task.Sources == null)
+                {
+                    continue;
+                }
+
+                foreach (Bounds bounds in task.Batches)
+                {
+                    stopWatch.Reset();
+                    stopWatch.Start();
+
+                    int totalTileCount = bounds.Size();
+                    int tileProgressCount = 0;
+
+                    List<Data> sources = BuildDataList(task.Sources, totalTileCount);
```

Comment:
if totalTileCount is 0 -> continue; (with log)

Reply:
No.